db/crud.go

```diff
@@ -629,9 +629,14 @@ func (db *Database) OnDemandImportForWrite(docid string, doc *document, body Bod
 	return nil
 }
 
+func (db *Database) Put(docid string, body Body) (newRevID string, err error) {
+	return db.PutRoundTrip(docid, body, false)
+}
+
 // Updates or creates a document.
 // The new body's BodyRev property must match the current revision's, if any.
-func (db *Database) Put(docid string, body Body) (newRevID string, err error) {
+// If roundTrip is true, this function blocks until the write has been cached.
+func (db *Database) PutRoundTrip(docid string, body Body, roundTrip bool) (newRevID string, err error) {
```

Comment:
Revisit whether it's a good idea to pass the request context in here, to avoid scenarios where the request timeout fires earlier than the retry timeout.